build.gradle.kts

```diff
@@ -593,20 +594,26 @@ tasks {
         }
     }
 
-    register<Exec>("generateWildflyBootableJar") {
+    // Simple function to invoke a maven goal, dependent on the os, with optional
+    register<Maven>("generateWildflyBootableJar") {
         dependsOn("war")
-        if (System.getProperty("os.name").lowercase(Locale.ROOT).contains("windows")) {
-            commandLine("./mvnw.cmd", "wildfly-jar:package")
-        } else {
-            commandLine("./mvnw", "wildfly-jar:package")
-        }
+        execGoal("wildfly-jar:package")
     }
 
-    register<Exec>("mavenClean") {
-        if (System.getProperty("os.name").lowercase(Locale.ROOT).contains("windows")) {
-            commandLine("./mvnw.cmd", "clean")
-        } else {
-            commandLine("./mvnw", "clean")
-        }
+    register<Maven>("mavenClean") {
+        execGoal("clean")
     }
 }
+
+@DisableCachingByDefault(because = "Gradle would require more information to cache this task")
+abstract class Maven : Exec() {
+    // Simple function to invoke a maven goal, dependent on the os, with optional
+    fun execGoal(goal: String, vararg args: String) = commandLine(
+            if (System.getProperty("os.name").lowercase(Locale.ROOT).contains("windows")) {
+                "./mvnw.cmd"
+            } else {
+                "./mvnw"
+            },
+            goal,
+            *args)
```

Comment:
Nice one! Nitpicking a bit, but the last `)` could be moved to the next line for better readability.

Reply:
fixed this